src/ol/interaction/Snap.js

```diff
@@ -288,25 +304,55 @@ class Snap extends PointerInteraction {
     return features;
   }
 
+  /**
+   * Checks if two snap data sets are equal.
+   * Compares the Y-coordinate of the `vertex` and the feature ID.
+   *
+   * @param {SnappedInfo} data1 The first snap data set.
+   * @param {SnappedInfo} data2 The second snap data set.
+   * @return {boolean} `true` if the data sets are equal, otherwise `false`.
+   * @private
+   */
+  areSnapDataEqual_(data1, data2) {
+    return (
+      data1.vertex[1] === data2.vertex[1] &&
+      data1.feature?.getId() === data2.feature?.getId()
```

Comment:
Features do not need an id, but the feature reference will be the same if the features are the same
```suggestion
      data1.feature === data2.feature
```